onnxruntime/test/providers/cpu/tensor/gather_elements_op_test.cc

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "gtest/gtest.h"
+#include "test/providers/provider_test_utils.h"
+
+namespace onnxruntime {
+namespace test {
+
+template <typename T>
+void RunTypedTest()
+{
+  // int32_t indices - axis 0
+  OpTester test1("GatherElements", 11);
+
+  test1.AddAttribute<int64_t>("axis", 0LL);
+  test1.AddInput<T>("data", {2, 3},
+                       {0, 1, 2, 3, 4, 5});
+  test1.AddInput<int32_t>("indices", {1, 2}, {0, 1});
+  test1.AddOutput<T>("output", {1, 2},
+                        {0, 4});
+  test1.Run();
+
+  // int32_t indices - axis 1
+  OpTester test2("GatherElements", 11);
+  test2.AddAttribute<int64_t>("axis", 1LL);
+  test2.AddInput<T>("data", {2, 2},
+                       {1, 2,
+                        3, 4});
+  test2.AddInput<int32_t>("indices", {2, 2},
+                         {0, 0,
+                          1, 0});
+  test2.AddOutput<T>("output", {2, 2},
+                        {1, 1,
+                         4, 3});
+  test2.Run();
+
+  
+  // int64_t indices - axis 1
+  OpTester test3("GatherElements", 11);
+  test3.AddAttribute<int64_t>("axis", 1LL);
+  test3.AddInput<T>("data", {2, 2},
+                       {1, 2,
+                        3, 4});
+  test3.AddInput<int64_t>("indices", {2, 2},
+                         {0, 0,
+                          1, 0});
+  test3.AddOutput<T>("output", {2, 2},
+                        {1, 1,
+                         4, 3});
+  test3.Run();
+
+  // negative indices - axis 1
+  OpTester test4("GatherElements", 11);
+  test4.AddAttribute<int64_t>("axis", 1LL);
+  test4.AddInput<T>("data", {2, 2},
+                    {1, 2,
+                     3, 4});
+  test4.AddInput<int64_t>("indices", {2, 2},
+                          {0, 0,
+                           -1, -1});
+  test4.AddOutput<T>("output", {2, 2},
+                     {1, 1,
+                      4, 4});
+  test4.Run();
+
+  // indices out of bounds
+  OpTester test5("GatherElements", 11);
+  test5.AddAttribute<int64_t>("axis", 1LL);
+  test5.AddInput<T>("data", {2, 2},
+                    {1, 2,
+                     3, 4});
+  test5.AddInput<int64_t>("indices", {2, 2},
+                          {0, 0,
+                           2, 2});
+  test5.AddOutput<T>("output", {2, 2},
+                     {1, 1,
+                      4, 4});
+  test5.Run(OpTester::ExpectResult::kExpectFailure, "GatherElements op: Value in indices must be within bounds [-2 , 1]");
+
+  // 3D input - axis 1
+  OpTester test6("GatherElements", 11);
```

Comment:
Can you add a test with axis = 2? 
I've seen a few cases where code that works for 0 and 1 breaks at 2 or more and it wasn't discovered until a test was added.

Reply:
Sure. Will do.

Reply:
Added test for axis = 2